docs/decisions/0011-fhirserver.md

```diff
@@ -0,0 +1,130 @@
+
+Pros:
+- With a RDBMS solution, data can be accessed via queries quickly, there is no need to manipulate data further for analytics purpose
+- The team will have full control of how the server works and can have more customized solutions
+
+Cons:
+- Resource cost for the developers will be high due to the infrastructure needed to create this service
+- Resource cost for the public health departments to understand how our data is structured because it is not standard
+- Using a non-standardized data structure may cause a lot of headaches when the product is used by multiple public health departments
+
+### Data Streams via Message Queues
+
+As a way to improve our solution to fit different use cases, we could use a messaging queue system such as AWS EventBridge to use the appropritate resource to process the request. For example, if a STLT would like to use FHIR, they can request that their data be processed in our FHIR server. If a different STLT does not mind using a non-standard solution, and wants quicker analytics, they can go to a RDBMS solution. 
+
+The way to achieve this is a messaging-queue system that sorts messages into different process pipelines so that one system can support decoupled solutions. Eventbridge is serverless so there is less maintenence required to stand up and maintain the resource. 
+
+[AWS EventBridge](https://aws.amazon.com/eventbridge/)
+
+### Which FHIR Server?
+
+#### Google Cloud FHIR Server
+
+Google Cloud FHIR service has tight integration with other GCP products such as BigQuery, Storage, Pub/Sub Integration
+
+Pros:
+- In GCP FHIR Server, you pay for what you use. This advantage would be key if demand is unpredictable or high variance. 
+- Access to tools such as BigQuery for data analytics. 
+- Supports all versions of FHIR
+
+Cons:
+- More expensive for storage and API calls
+
+#### AWS FHIR Works
+
+AWS FHIRWorks is Amazon's FHIR Server product that integrates with AWS services such as DynamoDB, Elasticsearch, Lambda, etc.
+
+Pros:
+- FHIR Works is the cheapest of the three options 
+- Integrates with natural language processing services for specific medical use cases.
+
+Cons:
+- Does not support certain key features such as patch calls, bundles, or conditionals
+
+#### Mircrosoft Azure FHIR Server
+
+Azure FHIR server is Azure's FHIR server product that allows developers to create FHIR applications using Azure products.
+
+Pros:
+- Since the CDC already uses Azure, using Azure FHIR server will be easier to integrate with their existing system. 
+- Azure FHIR server supports both NoSQL (CosmoDB) and SQL. Azure is pushing more features on to their SQL solution. 
+
+Cons:
+- If using CosmoDB, users must manage partitioning and set number of request units
+- Pay-for-what-you-might-use model. Currently Azure is running on a prepaid system which may be worse for unpredictable demand.
+- Certain search and API features are missing in Azure that is supported by GCP
+
+
+## Decision Outcome
```

Comment:
I would split this into two decisions:
* One for overall architecture
* One for specific platform for the reference implementation

Reply:
Done